src/search3_jobs.erl

```diff
@@ -0,0 +1,83 @@
+-module(search3_jobs).
+
+-export([
+    set_timeout/0,
+    build_search/3,
+    build_search_async/2
+]).
+
+-include("search3.hrl").
+
+set_timeout() ->
+    couch_jobs:set_type_timeout(?SEARCH_JOB_TYPE, 6 * 1000).
```

Comment:
Make sure check that your indexers, once they accept a job will keep updating at least once during a 6 second period. Otherwise there is a risk the jobs will be considered timed out and re-enqueued. You might have already checked for this but thought I'd mentioned it as something to keep an eye on.